--- Python-3.3.0/Lib/distutils/sysconfig.py	2012-09-29 10:00:28.000000000 +0200
+++ Python-3.3.0/Lib/distutils/sysconfig.py	2012-12-22 20:14:49.661337174 +0100
@@ -228,7 +228,7 @@ def customize_compiler(compiler):
             linker_exe=cc,
             archiver=archiver)
 
-        compiler.shared_lib_extension = so_ext
+        #compiler.shared_lib_extension = so_ext
 
 
 def get_config_h_filename():
