diff -up Python-3.2a1/Lib/test/test_imp.py.apply-our-changes-to-expected-shebang Python-3.2a1/Lib/test/test_imp.py
--- Python-3.2a1/Lib/test/test_imp.py.apply-our-changes-to-expected-shebang	2010-08-02 15:57:49.151002176 -0400
+++ Python-3.2a1/Lib/test/test_imp.py	2010-08-02 15:58:40.428010729 -0400
@@ -52,7 +52,7 @@ class ImportTests(unittest.TestCase):
         self.assertNotEqual(fp, None)
         self.assertEqual(fp.encoding, "iso-8859-1")
         self.assertEqual(fp.tell(), 0)
-        self.assertEqual(fp.readline(), '#!/usr/bin/env python3\n')
+        self.assertEqual(fp.readline(), '#!/usr/bin/python3.2\n')
         fp.close()
 
         fp, filename, info = imp.find_module("tokenize")
